### PR TITLE
sdf: refactor test package

### PR DESCRIPTION
**Pull request by ErniGH**
Simplify the test_package in according to our policies:

- Write the minimal test that tests the package (the compiler can find the correct includes, the static linker can resolve
symbols, the runtime linker can resolve symbols).
- Do not perform heavy logic in CI.
- Remove binary blob - these have not been allowed anywhere in the repository for a while now, it's time to drop them where they exist

### CODE REVIEW

File: recipes/sdf/all/test_package/test_package.c [hunk 1]
```diff
@@ -8,30 +8,8 @@
...
-        return 1;
-    }
-
-    sdfBuildDistanceField(dest_data, width, 2.0f, img_data, width, height, width);
-
-    free(dest_data);
-    stbi_image_free(img_data);
-
+    unsigned char* img_data = stbi_load("not a valid/path/fake_img.png", &width, &height, &bpp, 0);
+    printf("img_data: %p\n", img_data);
```
Comment (RubenRBS) [inline recipes/sdf/all/test_package/test_package.c:13]: It's probably UB to print a `NULL` pointer in this case, (https://stackoverflow.com/a/44996472) I'd keep the error message with the check in there, but make it return ok

**Comment by SpaceIm** [comment]
It was already a minimal test package. Your proposed test package doesn't even call any function of sdf :/

**Comment by ErniGH** [comment]
@SpaceIm Oh well noted. The test was very good and was actually quite small. The change is actually to remove the binary files from our test packages. 😄

File: recipes/sdf/all/test_package/test_package.c [hunk 2]
```diff
@@ -8,27 +8,17 @@
...
+    unsigned char* img_data = stbi_load("not a valid/path/fake_img.png", &width, &height, &bpp, 0);
+    if (img_data == NULL)
     {
-        stbi_image_free(img_data);
-        return 1;
+        printf("Test\n"); // This should be printed always
     }
 
-    sdfBuildDistanceField(dest_data, width, 2.0f, img_data, width, height, width);
+    int result = sdfBuildDistanceField(dest_data, width, 2.0f, img_data, width, height, width);
```
Comment (SpaceIm) [inline recipes/sdf/all/test_package/test_package.c:20]: You can't give a NULL pointer as img_data to this function. If you want to drop binary image from conan-center repo, you'll have to programatically create pixels in this img_data. And no need to keep stb dependency anymore.

**Review by uilianries — DISMISSED** [review]

**Review by SpaceIm — APPROVED** [review]

**Comment by SpaceIm** [comment]
`stb` should be removed from requirements in test package

**Review by RubenRBS — APPROVED** [review]

**Comment by SpaceIm** [comment]
There is still `stb` in conanfile of test_v1_package.

**Comment by RubenRBS** [comment]
Great catch @SpaceIm :)

**Review by RubenRBS — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Pull request merged**